x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedTimingStatsReporter.java

```diff
@@ -33,12 +33,15 @@
     private volatile DatafeedTimingStats currentTimingStats;
     /** Object used to persist current timing stats. */
     private final DatafeedTimingStatsPersister persister;
+    /** Whether or not timing stats will be persisted by the persister object. */
+    private boolean allowedPersisting;
```

Comment:
Maybe it should be `volatile`, as in production it will be set from a different thread to the one it's read from?

Reply:
Done.